src/main/java/org/tron/core/capsule/AccountCapsule.java

```diff
@@ -281,6 +380,40 @@ public boolean reduceAssetAmount(byte[] key, long amount) {
     return false;
   }
 
+  /**
+   * reduce asset amount.
+   */
+  public boolean reduceAssetAmountV2(byte[] key, long amount, Manager manager) {
+    if (manager.getDynamicPropertiesStore().getAllowSameTokenName() == 0) {
+      byte[] tokenName = key;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Store a copy of "key". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1647&issues=AWcW2bxfgA_rbE7wDiYV&open=AWcW2bxfgA_rbE7wDiYV)

[ISSUE_KEY]: AWcW2bxfgA_rbE7wDiYV